flake.nix

```diff
@@ -0,0 +1,55 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    nixpkgs-review.url = "github:Mic92/nixpkgs-review";
+  };
+
+  outputs =
+    {
+      self,
+      nixpkgs,
+      nixpkgs-review,
+      ...
+    }:
+
+    let
+      inherit (nixpkgs) lib;
+
+      eachSystem = f: lib.genAttrs systems (system: f (import nixpkgs { inherit system overlays; }));
+      systems = [
+        "x86_64-linux"
+        "aarch64-linux"
+        "x86_64-darwin"
+        "aarch64-darwin"
+      ];
+
+      overlays = [ self.overlays.nixpkgs-review ];
+    in
+
+    {
+      overlays.nixpkgs-review = final: prev: {
+        inherit (nixpkgs-review.packages.${final.system}) nixpkgs-review;
+      };
+
+      legacyPackages = eachSystem lib.id;
+
+      formatter = eachSystem (
+        pkgs:
+        pkgs.treefmt.withConfig {
+          settings = lib.mkMerge [
+            ./treefmt.nix
+            { _module.args = { inherit pkgs; }; }
```

Comment:
Is there a reason you're setting `pkgs` explicitly?

The default `pkgs` instance is the one passed to `pkgs.callPackage`: https://github.com/NixOS/nixpkgs/blob/92c0a34797dbe1eb63d839b5b807947732032d95/pkgs/by-name/tr/treefmt/lib.nix#L29 -- is that one different to the top-level `pkgs` your're using here?